tests/storage/test_event_push_actions.py

```diff
@@ -196,6 +196,13 @@ def _mark_read(stream: int, depth: int) -> None:
         _mark_read(10, 10)
         _assert_counts(0, 0)
 
+        _inject_actions(11, HIGHLIGHT)
+        _assert_counts(1, 1)
+        _mark_read(11, 11)
+        _assert_counts(0, 0)
+        _rotate(11)
+        _assert_counts(0, 0)
```

Comment:
Which assertion fails without this change: the final one on +204?

I an see that the changed query now selects fewer event_push_actions. So is the point that we don't present "old" info in new summaries? (I'm not even sure what a summary is or how highlights play into this.

Reply:
Yup, on 204.

The `event_push_summary` is a table that for each user/room has a count of all unread notifications, i.e. its a cache of `COUNT(*)` on `event_push_actions`. The rotation BG job is what iteratively updates `event_push_summary`, i.e. it periodically checks for new notifs and adds the new counts to `event_push_summary`. But we also need to recalculate the counts when we get a receipt. The bug here is that looked at the same range of `event_push_actions` to update the summary twice, once when handling receipts and then once when rotating as normal